_posts/2018-10-11-migrating-from-galen-to-backstop.md

```diff
@@ -0,0 +1,242 @@
+<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+And that is what we did: **we made it easy to write the tests, so that it was easy to migrate them** (make the change).
+
+That task consisted out of three steps:
+
+1. Simplifying test's structure. Eventually we came to the point where taking a screenshot of a page under a given URL looked like so:
+
+    <em class="snippet-description">Test suite with one test case</em>
+    ```javascript
+    // tests suite
+    module.exports = [
+        // single test
+        {
+            path: '/to-some-page',
+            label: 'label-for-that-page'
+        }
+    ];
+    ``` 
+
+1. Preparing basic, very abstract interactions (click, input text, scroll, wait for element, etc.). Let's say you wanted to input some text and submit the form, here is what you had to do:
+
+    <em class="snippet-description">Interactions usage example</em>
+    ```javascript
+    const { CLICK, INPUT_TEXT } = require('./interactions');   
+ 
+    module.exports = [
+        {
+            path: '/to-some-page-with-a-form',
+            label: 'label-for-that-page',
+            interactions: [
+                {   
+                    type: INPUT_TEXT,
+                    selector: '.some-form-field',
+                    text: 'Hello, world!'
+                },
+                {
+                    type: CLICK,
+                    selector: '.submit-button' 
+                }
+            ]
+        }
+    ];
+    ```
+
+1. And finally, writing the docs on how to migrate the tests, so that others could understand and follow.
+
+When all of the above were done, we split the task to few of the guys in the team and migrated all of the tests.
+
+Did it go well? Of course not! :)
+
+## Problems
+
+### Navigation on mobile-like devices
+
+One of the tests was about clicking on a menu item to drop it down to reveal a submenu. The trick was that this element behaved differently on phone (clicking was not resulting in redirection). Galen test had it done in a simple "click a selector"-way. It took us a while to find out that there was a hack included that had overridden the touch event: 
+
+<em class="snippet-description">Script disabling redirection for mobile devices</em>
+```javascript
+window.document.documentElement.ontouchend = {}
+```
+
+As puppeteer allows us to easily control the browser (Chrome in headless mode), we were able to order it: emulate phone. And we did that. And it worked. But other tests failed. After trying some solutions, we had to use the hack (mentioned above) eventually. Luckily, thanks to how we prepared all of the interaction, this script was hidden from the outside world in a form of a `DISLABLE_REDIRECT_ON_MOBILE` interaction. At least it was easy to use.
```

Comment:
These two short sentences going one after another: `And we did that. And it worked.`. Maybe replace it with something like: "We did that and it worked". This way you also avoid starting the sentence with `And`